mlflow/store/model_registry/file_store.py

```diff
@@ -0,0 +1,767 @@
+
+    def _get_registered_model(self, name):
+        model_path = self._get_registered_model_path(name)
+        if model_path is None:
+            raise MlflowException(
+                f"Could not find registered model with name {name}",
+                RESOURCE_DOES_NOT_EXIST,
+            )
+        return self.__get_registered_model_from_path(model_path)
+
+    def update_registered_model(self, name, description):
+        """
+        Update description of the registered model.
+
+        :param name: Registered model name.
+        :param description: New description.
+        :return: A single updated :py:class:`mlflow.entities.model_registry.RegisteredModel` object.
+        """
+        registered_model = self._get_registered_model(name)
+        updated_time = now()
+        registered_model.description = description
+        registered_model.last_updated_timestamp = updated_time
+        registered_model_dict = dict(registered_model)
+        del registered_model_dict["tags"]
+        write_yaml(
+            join(self.root_directory, name),
+            FileStore.META_DATA_FILE_NAME,
+            registered_model_dict,
+            overwrite=True,
+        )
+        return registered_model
+
+    def rename_registered_model(self, name, new_name):
+        """
+        Rename the registered model.
+
+        :param name: Registered model name.
+        :param new_name: New proposed name.
+        :return: A single updated :py:class:`mlflow.entities.model_registry.RegisteredModel` object.
+        """
+        model_path = self._get_registered_model_path(name)
+        if model_path is None:
+            raise MlflowException(
+                f"Could not find registered model with name {name}",
+                RESOURCE_DOES_NOT_EXIST,
+            )
+        registered_model = self.__get_registered_model_from_path(model_path)
+
+        new_meta_dir = self._get_registered_model_path(new_name)
+        if new_meta_dir is None:
+            updated_time = now()
+            registered_model.name = new_name
+            registered_model.last_updated_timestamp = updated_time
+            mkdir(new_meta_dir)
+            write_yaml(new_meta_dir, FileStore.META_DATA_FILE_NAME, dict(registered_model))
+            model_versions = self._list_model_versions_under_path(model_path)
+            for mv in model_versions:
+                mv.name = new_name
+                mv.last_updated_timestamp = updated_time
+                new_model_version_dir = join(new_meta_dir, f"version-{mv.version}")
+                mkdir(new_model_version_dir)
+                model_version_dict = dict(mv)
+                del model_version_dict["tags"]
+                write_yaml(new_model_version_dir, FileStore.META_DATA_FILE_NAME, model_version_dict)
+                if mv.tags is not None:
+                    for tag in mv.tags:
+                        self.set_model_version_tag(new_name, mv.version, tag)
+            shutil.rmtree(model_path)
```

Comment:
The loop performance is bad. We can remove the loop by:
1. Avoid saving "mv.name" in model version meta file
we can check the parent diretory name of the model version meta file, to get the register-model name.
2. Avoid update "last_updated_timestamp" field in mv meta file when register model name is updated. Instead, when we want to read `mv.last_updated_timestamp`, we read last_updated_timestamp of both "registered model" and this "model version", and return the bigger timestamp

Reply:
I think changing to this logic will make other processes more complicated: for example, whenever you do updates to the model version, and it returns a `ModelVersion` object, we need to go over this logic to fetch the upper folder name and timestamp; Besides, if we save extra field in registered_model meta file, we need to exclude it when constructing the object from meta file; 
I do agree that saving `mv.name` in the registered model only makes sense, but this `rename_registered_model` looks like a relative less frequent move, WDYT?